pytensor/tensor/subtensor.py

```diff
@@ -186,8 +219,8 @@ def get_idx_list(inputs, idx_list):
 
 
 def get_canonical_form_slice(
-    theslice: slice | Variable, length: Variable
-) -> tuple[Variable, int]:
+    theslice: slice | Variable, length: ScalarVariable
```

Comment:
I think you can more precisely overload this function as well, you know if the input is a `ScalarVariable` or a python `slice`, and the first output type should be the same